tgui/packages/tgui/interfaces/FlockPanel.js

```diff
@@ -0,0 +1,324 @@
+        return (
+          <Stack.Item key={enemy.ref}>
+            <Stack>
+              {/* name and remove button */}
+              <Stack.Item width="30%">
+                <Section height="100%">
+                  {enemy.name}
+                </Section>
+              </Stack.Item>
+              {/* area and jump button */}
+              <Stack.Item grow={1}>
+                <Section height="100%">
+                  <Stack>
+                    <Stack.Item grow={1}>
+                      <b>{enemy.area}</b>
+                    </Stack.Item>
+
+                  </Stack>
+                </Section>
+              </Stack.Item>
+              <Stack.Item>
+                <Section height="100%">
+                  <Stack>
+                    <Stack.Item>
+                      <Button icon="times" onClick={() => act('remove_enemy', { 'origin': enemy.ref })} >
+                        Remove
+                      </Button>
+                    </Stack.Item>
+                    <Stack.Item>
+                      <Button onClick={() => act('jump_to', { 'origin': enemy.ref })} >
+                        Jump
+                      </Button>
+                    </Stack.Item>
+                  </Stack>
+                </Section>
+              </Stack.Item>
+
+            </Stack>
+          </Stack.Item>
+        );
+      })}
+    </Stack>
+  );
+};
+
+export const FlockPanel = (props, context) => {
+  const { data, act } = useBackend(context);
+  const [tabIndex, setTabIndex] = useLocalState(context, 'tabIndex', 1);
+  const [sortBy, setSortBy] = useLocalState(context, 'sortBy', 'resources');
+  const {
+    vitals,
+    partitions,
+    drones,
+    structures,
+    enemies,
+  } = data;
+  return (
+    <Window
+      theme="flock"
+      title={"Flockmind " + vitals.name}
+      width={600}
+      height={450}
+    >
+      <Window.Content scrollable>
+        <Tabs>
+          <Tabs.Tab
+            selected={tabIndex === 1}
+            onClick={() => setTabIndex(1)}>
+            Drones {"(" + drones.length + ")"}
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={tabIndex === 2}
+            onClick={() => setTabIndex(2)}>
+            Partitions {"(" + partitions.length + ")"}
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={tabIndex === 3}
+            onClick={() => setTabIndex(3)}>
+            Structures {"(" + structures.length + ")"}
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={tabIndex === 4}
+            onClick={() => setTabIndex(4)}>
+            Enemies {"(" + enemies.length + ")"}
+          </Tabs.Tab>
+        </Tabs>
+
+        {tabIndex === 1
+        && (
+          <Box>
+            <Dropdown
+              options={["name", "health", "resources", "area"]}
+              selected="resources"
+              onSelected={(value) => setSortBy(value)}
+            />
+            <FlockDrones drones={drones} sortBy={sortBy} />
+          </Box>
```

Comment:
A little weird that this isn't driven by local state on the FlockDrones component and is instead passed here, but I don't know much about tgui.

Reply:
Yeah, it's passed in so I don't have to put the dropdown into FlockDones mostly just so it matches the other Flock components, no idea which is better practice, though it should probably at least be called "sortDrones" or something if it is left outside.

Reply:
OH WELL WHAT COULD POSSIBLY GO WRONG